src/shell/autoexec.cpp

```diff
@@ -260,6 +260,39 @@ class AutoExecModule final : public Module_base {
 	void ReMountDirAsDriveC(const std::string& directory);
 
 	void AddMessages();
+
+	struct AutoMountSettings {
+		std::string override_drive = {};
+		std::string type           = {};
+		std::string label          = {};
+		bool readonly              = false;
+		std::string path           = {};
+		bool verbose               = false;
+	};
+
+	// Specify a 'Drive' config object with allowed key and value types
+	static std::unique_ptr<Config> SpecifyDriveConf();
+
+	// Parse a 'Drive' config file and return object with allowed key and
+	// value types
+	static std::optional<AutoMountSettings> ParseDriveConf(const std_fs::path& conf_path);
+
+	// Build a command to mount a drive from a folder, based on the
+	// provided settings
+	static std::string BuildAutoMountFolderCommand(
```

Comment:
Style: I would prefer `AutoMountDirectoryCommand` name (same for other identifiers with _folder_ in them). All the APIs we use (`POSIX`, `std::fs`, etc.) use a _directory_ term (not the _folder_ term), same with almost every single piece of currently supported guest software.

Reply:
Good point, please only use "directory" in the code.